bin/compile

```diff
@@ -79,19 +79,32 @@ heading() {
 APP_DIR=$(cd $(dirname "${BASH_SOURCE[0]}"); cd ..; pwd)
 BIN_DIR="$APP_DIR/vendor/bin"
 
-decrypt_output=$(echo $EJSON_PRIVATE_KEY | $BIN_DIR/ejson decrypt --key-from-stdin "$APP_DIR/$EJSON_FILE" 2>&1)
+exports=$(echo $EJSON_PRIVATE_KEY | \
+  $BIN_DIR/ejson decrypt --key-from-stdin "$APP_DIR/$EJSON_FILE" 2>&1 | \
+  $BIN_DIR/jq -r 'to_entries|map("export \(.key)=\(.value|tojson)")[]' 2>&1 | \
+  grep -v '^export _public_key=' | \
+  while IFS=$'\n' read -r line
+  do
+    echo $line
+  done)
+
 if [ $? -eq 0 ]; then
-  exports=$(echo $decrypt_output | $BIN_DIR/jq -r "to_entries|map(\"\(.key)=\(.value|tostring)\")|.[]" | grep -v '^_public_key=')
-  for env_var in $exports; do
-    export $env_var
-  done
+  eval "$exports"
 else
-  heading $decrypt_output
+  heading "Decryption failed"
+
+  # Don't exit 1 here to allow heroku ps:exec to work
+  export EJSON_DECRYPT_FAILED=true
 fi
 EOP
 chmod +x $BUILD_DIR/.profile.d/export_ejson_secrets.sh
 
 export EJSON_PRIVATE_KEY=$(cat "$ENV_DIR/EJSON_PRIVATE_KEY")
 
 heading "Sourcing environment variables on compile"
+export HEROKU_COMPILE=true
```

Comment:
What's this env var about? What is the meaning of "compile" in the line above? The slug build phase?

Reply:
Removed, this wasn't in use. Yeah by compile it meant the time the slug is being built, not the time the app is starting up (for the profile.d script).